src/language-js/print/type-annotation.js

```diff
@@ -291,16 +291,21 @@ function printTupleType(path, options, print) {
   const types = node[typesField];
   const isNonEmptyTuple = isNonEmptyArray(types);
   const bracketsDelimiterLine = isNonEmptyTuple ? softline : "";
+  const openBracket = "[";
+  const closeBracket = "]";
+  if (!isNonEmptyTuple) {
+    return printEmptyArrayLike(path, openBracket, closeBracket, options);
+  }
   return group([
-    "[",
+    openBracket,
     indent([
       bracketsDelimiterLine,
       printArrayItems(path, options, typesField, print),
     ]),
     ifBreak(isNonEmptyTuple && shouldPrintComma(options, "all") ? "," : ""),
     printDanglingComments(path, options, /* sameIndent */ true),
     bracketsDelimiterLine,
-    "]",
```

Comment:
This part looks refactorable. You checked `isNonEmptyTuple` in the `if` above, so you can remove the same check and `printDanglingComments` from here. Also double negation in `if (!isNonEmptyTuple)` doesn't look very good.

Reply:
[f01a6ca](https://github.com/prettier/prettier/pull/13608/commits/f01a6cacb8358431380c72e762da37d1319eecee)